src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1820,8 +1820,8 @@ public void writeStyle( StyleInfo info, Style style ) throws IOException {
      */
     public void writeStyle( StyleInfo info, Style style, boolean format) throws IOException {
         synchronized ( styleCache ) {
-            File styleFile = dataDir().findOrCreateStyleSldFile(info);
-            BufferedOutputStream out = new BufferedOutputStream( new FileOutputStream( styleFile ) );
+            Resource styleFile = dataDir().style(info);
+            BufferedOutputStream out = new BufferedOutputStream( styleFile.out() );
```

Comment:
@bencaradocdavies If you check the history this line wasn't actually last changed by me. But yes, I will admit that you will see this in some of the much much older code that I wrote for GeoServer and GeoTools but I stopped using that style quite some time ago. I guess it is a good thing that I don't really write code for those projects anymore. You must be able to sleep a lot better at night knowing i am no longer polluting the code base. In my defence though before you proceed to blame me for every mis-formatting in the code base like you are doing on this pull request I would point out that (a) there were not any formal style guidelines when I was active and (b) there are others who are still active on the project who use this style much more than I do (at least from code I have seen).